### PR TITLE
Did some refactoring and renaming of local variables

### PR DESCRIPTION
**Pull request by araspaas**
Refactored some code. made some helper methods. Increased run time on spec harness though.

### CODE REVIEW

**Review by robertorodriguez12 — COMMENTED** [review]
Nice work here Austin!

**Review by robertorodriguez12 — APPROVED** [review]

File: lib/team_stats.rb [hunk 1]
```diff
@@ -12,81 +12,94 @@ def team_info(team_id)
...
+  def find_all_games_by_given_team_id(team_id)
+    @games.find_all {|game| game.home_team_id == team_id || game.away_team_id == team_id}
   end
 
   def favorite_opponent(team_id)
-    find_by_given_team_id = @games.find_all {|game| game.home_team_id == team_id || game.away_team_id == team_id}
-    x = find_by_given_team_id.map {|game| game.game_id}
-    y = @game_teams.find_all {|game| x.include?(game.game_id) && game.team_id != team_id}
-    sorted_by_team_id = y.group_by {|game| game.team_id}
+    find_game_ids = find_all_games_by_given_team_id(team_id).map {|game| game.game_id}
```
Comment (araspaas) [inline lib/team_stats.rb:79]: Trying to break this into a helper method makes the spec run very slow. around 50 seconds. so thats why i didn't refactor beyond just the variable name

File: lib/team_stats.rb [hunk 2]
```diff
@@ -12,81 +12,94 @@ def team_info(team_id)
...
+    @teams.find {|team| team.team_id == the_team}.team_name
   end
-
+  
   def rival(team_id)
-    find_by_given_team_id = @games.find_all {|game| game.home_team_id == team_id || game.away_team_id == team_id}
-    x = find_by_given_team_id.map {|game| game.game_id}
-    y = @game_teams.find_all {|game| x.include?(game.game_id) && game.team_id != team_id}
-    sorted_by_team_id = y.group_by {|game| game.team_id}
+    find_game_ids = find_all_games_by_given_team_id(team_id).map {|game| game.game_id}
```
Comment (araspaas) [inline lib/team_stats.rb:93]: same here as above

File: lib/team_stats.rb [hunk 3]
```diff
@@ -12,81 +12,94 @@ def team_info(team_id)
...
+  def team_games_grouped_by_season(team_id)
+    find_home_or_away(team_id).group_by {|game| game.season}
+  end
+
   def best_season(team_id)
-    team_wins = @game_teams.find_all {|game| game.team_id == team_id && game.result == "WIN"}
-    winning_game_ids = team_wins.map {|game| game.game_id}
-    team_games = @games.find_all {|game| game.away_team_id == team_id || game.home_team_id == team_id}
-    team_games_grouped_by_season = team_games.group_by {|game| game.season}
+    team_games_grouped_by_season = find_home_or_away(team_id).group_by {|game| game.season}
```
Comment (dcoleman21) [inline lib/team_stats.rb:32]: Nice job here!

File: lib/team_stats.rb [hunk 4]
```diff
@@ -12,81 +12,94 @@ def team_info(team_id)
...
 
   def worst_season(team_id)
-    team_wins = @game_teams.find_all {|game| game.team_id == team_id && game.result == "WIN"}
-    winning_game_ids = team_wins.map {|game| game.game_id}
-    team_games = @games.find_all {|game| game.away_team_id == team_id || game.home_team_id == team_id}
-    team_games_grouped_by_season = team_games.group_by {|game| game.season}
+    team_games_grouped_by_season = find_home_or_away(team_id).group_by {|game| game.season}
     team_games_grouped_by_season.each do |season, games_array|
-      wins = games_array.find_all {|game| winning_game_ids.include?(game.game_id)}.count
+      wins = games_array.find_all {|game| winning_game_ids(team_id).include?(game.game_id)}.count
```
Comment (dcoleman21) [inline lib/team_stats.rb:43]: Ah, the argument goes here!

File: lib/team_stats.rb (hunk 1 above)
Comment (dcoleman21) [reply lib/team_stats.rb:79]: Hey I think that is great. You tried it and it didn't improve the performance. Nice job breaking all of these out.

File: lib/team_stats.rb [hunk 5]
```diff
@@ -12,81 +12,94 @@ def team_info(team_id)
...
+    find_game_ids = find_all_games_by_given_team_id(team_id).map {|game| game.game_id}
+    find_given_ids_opponents = @game_teams.find_all {|game| find_game_ids.include?(game.game_id) && game.team_id != team_id}
+    sorted_by_team_id = find_given_ids_opponents.group_by {|game| game.team_id}
     result = sorted_by_team_id.transform_values do |game|
-      w = game.find_all do |game|
+      count_of_wins = game.find_all do |game|
         (game.result == "WIN")
       end.count
-      (w.to_f / game.count).round(2)
+      (count_of_wins.to_f / game.count).round(2)
```
Comment (dcoleman21) [inline lib/team_stats.rb:100]: I like the variable names you picked

**Review by dcoleman21 — COMMENTED** [review]
Nice work here Austin. I think that those few methods that ran slower when broken down are good to stay as they are. We can look at other ways to refactor and see what we can come up with as well.

**Pull request merged**